src/stream.rs

```diff
@@ -219,6 +267,79 @@ impl<'a> Stream<'a> {
         Ok(())
     }
 
+    pub fn parse_ident(&mut self) -> Result<&'a str, Error> {
+        let start = self.pos();
+
+        if self.curr_byte() == Ok(b'-') {
+            self.advance(1);
+        }
+
+        let mut iter = self.chars();
+        if let Some(c) = iter.next() {
+            if c.is_name_start() {
+                self.advance(c.len_utf8());
+            } else {
+                return Err(Error::InvalidIdent);
+            }
+        }
+
+        for c in iter {
+            if c.is_name_char() {
+                self.advance(c.len_utf8());
+            } else {
+                break;
+            }
+        }
+
+        if start == self.pos() {
+            return Err(Error::InvalidIdent);
+        }
+
+        let name = self.slice_back(start);
+        Ok(name)
+    }
+
+    pub fn consume_ascii_ident(&mut self) -> &'a str {
+        let start = self.pos;
+        self.skip_bytes(|_, c| c.is_ascii_ident());
+        self.slice_back(start)
+    }
+
+    pub fn parse_string(&mut self) -> Result<&'a str, Error> {
+        // Check for opening quote.
+        let quote = self.curr_byte()?;
+        if quote == b'\'' || quote == b'"' {
```

Comment:
I think we can return immediately here if there are no quotes. No need for a nested code.

Reply:
Oh, I see now. This was copy-pasted from cssparser, so I reviewed my own code...